src/ppom-controller.ts

```diff
@@ -589,35 +590,13 @@ export class PPOMController extends BaseControllerV2<
         ),
       true,
     );
-    if (response?.status !== 200) {
-      throw new Error(`Failed to fetch file with url: ${url}`);
+    if (response?.status === 304) {
+      cached = true;
     }
-    return response;
-  }
-
-  /*
-   * Function sends a HEAD request to version info file and compares the ETag to the one saved in controller state.
-   * If ETag is not changed we can be sure that there is not change in files and we do not need to fetch data again.
-   */
-  async #checkIfVersionInfoETagChanged(url: string): Promise<boolean> {
-    const headResponse = await this.#getAPIResponse(
-      url,
-      {
-        headers: versionInfoFileHeaders,
-      },
-      'HEAD',
-    );
-
-    const { versionFileETag } = this.state;
-    if (headResponse.headers.get('ETag') === versionFileETag) {
-      return false;
+    if (!response?.status || response?.status < 200 || response?.status > 399) {
+      throw new Error(`Failed to fetch file with url: ${url}`);
     }
-
-    this.update((draftState) => {
-      draftState.versionFileETag = headResponse.headers.get('ETag');
-    });
-
-    return true;
+    return { cached, response };
```

Comment:
We can update the TS #getAPIResponse return value to be more specific. Currently it is Promise<any>